gradle-baseline-java/src/main/groovy/com/palantir/baseline/tasks/CheckImplicitDependenciesTask.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.baseline.tasks;
+
+import com.google.common.collect.Sets;
+import com.google.common.collect.Streams;
+import com.palantir.baseline.plugins.BaselineExactDependencies;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.GradleException;
+import org.gradle.api.artifacts.Configuration;
+import org.gradle.api.artifacts.ModuleVersionIdentifier;
+import org.gradle.api.artifacts.ResolvedArtifact;
+import org.gradle.api.artifacts.ResolvedDependency;
+import org.gradle.api.file.FileCollection;
+import org.gradle.api.provider.ListProperty;
+import org.gradle.api.provider.Property;
+import org.gradle.api.provider.Provider;
+import org.gradle.api.provider.SetProperty;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.InputFiles;
+import org.gradle.api.tasks.TaskAction;
+
+public class CheckImplicitDependenciesTask extends DefaultTask {
+
+    private final ListProperty<Configuration> dependenciesConfigurations;
+    private final Property<FileCollection> sourceClasses;
+    private final SetProperty<String> ignore;
+
+    public CheckImplicitDependenciesTask() {
+        setGroup("Verification");
+        setDescription("Ensures all dependencies are explicitly declared, not just transitively provided");
+        dependenciesConfigurations = getProject().getObjects().listProperty(Configuration.class);
+        dependenciesConfigurations.set(Collections.emptyList());
+        sourceClasses = getProject().getObjects().property(FileCollection.class);
+        ignore = getProject().getObjects().setProperty(String.class);
+        ignore.set(Collections.emptySet());
+    }
+
+    @TaskAction
+    public final void checkImplicitDependencies() {
+        Set<ResolvedDependency> declaredDependencies = dependenciesConfigurations.get().stream()
+                .map(Configuration::getResolvedConfiguration)
```

Comment:
Could we avoid resolvingConfigurations (and potentially compiling the whole repo) by looking at the incoming artifacts instead?